File: PythonDatabaseCompatibility/tableStats.py
import json

''' 
    Holds information regarding table and column types
    Assumes create table format is Create Table <table name>.
    If this changes, then modification required.
    '''

class TableStats:
    
    def __init__(self, sqlStr):
        self.dateTimeFlag = None
        self.sqlStmt = sqlStr
        self.tblName = sqlStr.split()[2]
        
        if 'DATETIME YEAR TO SECOND' in sqlStr:
            self.dateTimeFlag = "seconds"
        elif 'DATETIME YEAR TO FRACTION(1)' in sqlStr:
            self.dateTimeFlag = "fraction(1)"
        elif 'DATETIME YEAR TO FRACTION(2)' in sqlStr:
            self.dateTimeFlag = "fraction(2)"
        elif 'DATETIME YEAR TO FRACTION(3)' in sqlStr:
            self.dateTimeFlag = "fraction(3)"
        elif 'DATETIME YEAR TO FRACTION(4)' in sqlStr:
            self.dateTimeFlag = "fraction(4)"
        else:
            self.dateTimeFlag = "fraction(5)"        
        

File: RubyDatabaseCompatibility/tableStats.rb
class TableStats
	attr_accessor :dateTimeFlag, :sqlStr, :tblName

	def initialize(sqlStr)
		@dateTimeFlag = nil
		@sqlStmt = sqlStr
		@tblName = sqlStr.split()[2]

		if sqlStr.include?("DATETIME YEAR TO SECOND")
			@dateTimeFlag = "seconds"
		elsif sqlStr.include?("DATETIME YEAR TO FRACTION(1)")
			@dateTimeFlag = "fraction(1)"
		elsif sqlStr.include?("DATETIME YEAR TO FRACTION(2)")
			@dateTimeFlag = "fraction(2)"
		elsif sqlStr.include?("DATETIME YEAR TO FRACTION(3)")
			@dateTimeFlag = "fraction(3)"
		elsif sqlStr.include?("DATETIME YEAR TO FRACTION(4)")
			@dateTimeFlag = "fraction(4)"
		else
			@dateTimeFlag = "fraction(5)"
		end
	end
end

